Author: Tiago Vignatti <tiago.vignatti@intel.com>

There is currently a mismatch in the way GL headers are picked up between
third_party/khronos and third_party/mesa which can cause crashes in some
architectures due to different pointer sizes.

Upstream bug: https://code.google.com/p/chromium/issues/detail?id=266310

--- src/third_party/mesa/src/include/EGL/eglplatform.h
+++ src/third_party/mesa/src/include/EGL/eglplatform.h
@@ -104,6 +104,12 @@ typedef struct ANativeWindow        *EGLNativeWindowType;
 typedef struct egl_native_pixmap_t  *EGLNativePixmapType;
 typedef void                        *EGLNativeDisplayType;
 
+#elif defined(USE_OZONE)
+
+typedef intptr_t EGLNativeDisplayType;
+typedef intptr_t EGLNativeWindowType;
+typedef intptr_t EGLNativePixmapType;
+
 #elif defined(__unix__)
 
 #ifdef MESA_EGL_NO_X11_HEADERS
